src/factories/workflow-specific/PIM_WorkflowFactory_v1.sol

```diff
@@ -0,0 +1,291 @@
+    // Modifiers
+
+    modifier onlyPimFeeRecipient(address fundingManager) {
+        if (_msgSender() != _pimFeeRecipients[fundingManager]) {
+            revert PIM_WorkflowFactory__OnlyPimFeeRecipient();
+        }
+        _;
+    }
+
+    //--------------------------------------------------------------------------
+    // Constructor
+
+    constructor(
+        address _orchestratorFactory,
+        address _owner,
+        address _trustedForwarder
+    ) Ownable(_owner) ERC2771Context(_trustedForwarder) {
+        orchestratorFactory = _orchestratorFactory;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IPIM_WorkflowFactory_v1
+    function createPIMWorkflow(
+        IOrchestratorFactory_v1.WorkflowConfig memory workflowConfig,
+        IOrchestratorFactory_v1.ModuleConfig memory paymentProcessorConfig,
+        IOrchestratorFactory_v1.ModuleConfig[] memory moduleConfigs,
+        IPIM_WorkflowFactory_v1.PIMConfig memory PIMConfig
+    )
+        external
+        returns (IOrchestrator_v1 orchestrator, ERC20Issuance_v1 issuanceToken)
+    {
+        // deploy issuance token
+        issuanceToken = new ERC20Issuance_v1(
+            PIMConfig.issuanceTokenParams.name,
+            PIMConfig.issuanceTokenParams.symbol,
+            PIMConfig.issuanceTokenParams.decimals,
+            PIMConfig.issuanceTokenParams.maxSupply,
+            address(this) // assigns owner role to itself initially to manage minting rights temporarily
+        );
+
+        // assemble fundingManager config, authorizer config and deploy orchestrator
+        IOrchestratorFactory_v1.ModuleConfig memory fundingManagerConfig =
+        IOrchestratorFactory_v1.ModuleConfig(
+            PIMConfig.fundingManagerMetadata,
+            abi.encode(
+                address(issuanceToken),
+                PIMConfig.bcProperties,
+                PIMConfig.collateralToken
+            )
+        );
+        IOrchestratorFactory_v1.ModuleConfig memory authorizerConfig =
+        IOrchestratorFactory_v1.ModuleConfig(
+            PIMConfig.authorizerMetadata, abi.encode(address(this))
+        );
+        orchestrator = IOrchestratorFactory_v1(orchestratorFactory)
+            .createOrchestrator(
+            workflowConfig,
+            fundingManagerConfig,
+            authorizerConfig,
+            paymentProcessorConfig,
+            moduleConfigs
+        );
+
+        // get bonding curve / funding manager
+        address fundingManager = address(orchestrator.fundingManager());
+
+        // make sure that curve configuration doesn't allow a negative static price since would break the math
+        if (IBondingCurveBase_v1(fundingManager).getStaticPriceForBuying() < 0)
+        {
+            revert
+                IPIM_WorkflowFactory_v1
+                .PIM_WorkflowFactory__InvalidConfiguration();
+        }
```

Comment:
Here is the "input validation"

Reply:
we need to enforce (staticPrice > 0). The function returns a uint, so the value wouldn't ever be negative anyway

Reply:
Dang, I was assuming that it returns `int`. In this case I will have to reimplement/generalize a `pure` version of getStaticPrice after all..